openff/toolkit/topology/molecule.py

```diff
@@ -730,161 +744,38 @@ class VirtualSite(Particle):
-                )
-
-        # Perform type-checking
-        # for atom in atoms:
-        #     assert isinstance(atom, Atom)
-        # for atom_index in range(len(atoms) - 1):
-        #     assert atoms[atom_index].molecule is atoms[atom_index + 1].molecule
-        # assert isinstance(atoms[1].molecule, FrozenMolecule)
-
-        if sigma is None:
-            self._sigma = None
-        else:
-            assert hasattr(sigma, "unit")
-            assert unit.angstrom.is_compatible(sigma.unit)
-            self._sigma = sigma.in_units_of(unit.angstrom)
-
-        if epsilon is None:
-            self._epsilon = None
-        else:
-            assert hasattr(epsilon, "unit")
-            assert (unit.kilojoule_per_mole).is_compatible(epsilon.unit)
-            self._epsilon = epsilon.in_units_of(unit.kilojoule_per_mole)
-
-        if charge_increments is None:
-            self._charge_increments = None
-        else:
-            for ci in charge_increments:
-                assert hasattr(ci, "unit")
-                assert unit.elementary_charges.is_compatible(ci.unit)
-            self._charge_increments = [
-                ci.value_in_unit(unit.elementary_charges) for ci in charge_increments
-            ] * unit.elementary_charges
-
-        self._atoms = list()
-
-        for atom in atoms:
-            atom.add_virtual_site(self)
-            self._atoms.append(atom)
-        self._molecule = atoms[0].molecule
-
-        self._name = name
+        self._molecule: Optional["Molecule"] = None
+        self._name: str = name
 
-        if orientations is None:
-            ornt = [tuple(atom.molecule_atom_index for atom in atoms)]
-            self._orientations = ornt
-            self._particles = {ornt[0]: VirtualParticle(self, ornt[0])}
+        if len(orientations) > 0 and type(orientations[0]) is int:
+            orientations = [tuple(orientations)]
         else:
-            ornt = None
-            if type(orientations[0]) is int:
-                ornt = [tuple(orientations)]
-            else:
-                ornt = [tuple(x) for x in orientations]
-            self._orientations = ornt
-            self._particles = dict(
-                {order: VirtualParticle(self, order) for order in ornt}
-            )
-
-        # Subclassing makes _type unnecessary
-        # self._type = None
-        # TODO: Validate site types against allowed values
+            orientations = [tuple(x) for x in orientations]
 
-        # self._weights = np.array(weights) # make a copy and convert to array internally
+        self._orientations: List[Tuple[int, ...]] = orientations
 
     def __eq__(self, other):
         if not issubclass(type(other), VirtualSite):
             return False
         if self.type != other.type:
             return False
         same_name = self.name == other.name
-        same_indices = self.atoms == other.atoms
+        same_indices = sorted(self.orientations) == sorted(other.orientations)
```

Comment:
Sort the orientation list in case one v-site has `[(0, 1), (1, 0)]` and another `[(1, 0), (0, 1)]` which yield the same v-site particles